### ISSUE TITLE
PROJECT_PATTERN Excludes "+"

### ISSUE THREAD

**Issue by danieldjewell** [uranusjr/packaging-dists#8]
The lines below exclude the "+" character from a valid name - I'm unsure if there is a PEP actually specifying valid characters, however: the "+" is often in filenames especially for wheels built from git repos. There are also several packages on Pypi with a "+" in the filename. (see https://pypi.org/pypi/docky/json) for a random example. 

https://github.com/uranusjr/packaging-dists/blob/726d132c0a2cc669a04fd0f0f3c567307f233fd9/src/packaging_dists/__init__.py#L28-L30

**Comment by uranusjr** [comment]
This is probably a mistake, the variable name indicates it is only supposed to apply to the “project name” part,  like this:

```python
try:
    n, v, *b, python, abi, platform = stem.split("-")
except ValueError:
    raise InvalidWheel(filename)
parsed_project = canonicalize_name(n)
if not PROJECT_PATTERN.match(parsed_project):
    raise InvalidWheel(filename)       
... 
```

**Comment by uranusjr** [comment]
Fixed in 7ce58f441e13158efa1957c83275c88029d8848c. I’ve released version 0.3 for this.
